src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,118 @@
+     * @throws FileNotFoundException if the program cannot find the file then it will throw the exception.
+     * @return a taskList object that tasks in the txt file is added into.
+     */
+    public TaskList readTasks(TaskList taskList) throws FileNotFoundException {
+        File f = new File(dirPath);
+        if ( ! f.exists()){
+            existDir = f.mkdir();
+        }
+        try{
+            f = new File(filePath);
+            if ( ! f.exists()){
+                existFile = f.createNewFile();
+            }
+        } catch (IOException e){
+            throw new FileNotFoundException();
+        }
+
+
+        Scanner s = new Scanner(f);
+        DateTimeFormatter df = DateTimeFormatter.ofPattern("MMM dd yyyy HH:mm");
+        while (s.hasNext()) {
+            String info = s.nextLine();
+            boolean status = Integer.parseInt(info.substring(0,1)) == 1;
+            String type = info.substring(3,4);
+
+            if (type.equals("T")){
+                String name = info.substring(5);
+                ToDo todo = new ToDo(name,status);
+                taskList.addTasks(todo);
+            } else if(type.equals("D")){
+                int endNameIndex = info.indexOf("(");
+                int endTimeIndex = info.length() - 1;
+                String name = info.substring(5 , endNameIndex - 1);
+                String by = info.substring(endNameIndex + 5, endTimeIndex);
+                LocalDateTime byTime = LocalDateTime.parse(by,df);
+                Deadline deadline = new Deadline(name, byTime, status);
+                taskList.addTasks(deadline);
+            } else if(type.equals("E")){
+                int endNameIndex = info.indexOf("(");
+                int endTimeIndex = info.length() - 1;
+                String name = info.substring(5 , endNameIndex - 1);
+                String at = info.substring(endNameIndex + 5, endTimeIndex);
+                LocalDateTime atTime = LocalDateTime.parse(at,df);
+                Event event = new Event(name, atTime, status);
+                taskList.addTasks(event);
+            }
+        }
+        return taskList;
+    }
+
+
+    /**
+     * The function will save the tasks immediately after the program terminates into the txt file into the
+     * path that the user specified.
+     *
+     * @param taskList The taskList that will be used in the program.
+     * @throws IOException if the program meets IO problem.
+     */
+    public void saveTasks(TaskList taskList) throws IOException {
+        FileWriter fw = new FileWriter(filePath, false);
+
+        StringBuilder builder = new StringBuilder();
+        if (taskList.getNumOfTasks() == 0) {
+            String textToAppend = builder.toString();
+            fw.write(textToAppend);
+            fw.close();
+        } else{
+            for (Task task:taskList.getTasks()){
+                int status = task.getStatus() ? 1 : 0;
```

Comment:
Good one line logic here. Perhaps using "taskStatus" instead of "status" would make it clearer to the reader?